OWML.ModHelper.Menus/ModComboInput.cs

```diff
@@ -0,0 +1,140 @@
+            noButton.transform.parent.gameObject.SetActive(false);
+            _layoutGroup = Button.LayoutGroup;
+            ((RectTransform)_layoutGroup.transform).sizeDelta = new Vector2(((RectTransform)Button.Button.transform.parent).sizeDelta.x * 2, ((RectTransform)Button.Button.transform.parent).sizeDelta.y);
+
+            var layoutGroup = Button.Button.transform.parent.parent.GetComponent<HorizontalLayoutGroup>();
+            layoutGroup.childControlWidth = true;
+            layoutGroup.childForceExpandWidth = true;
+            Button.Button.transform.parent.GetComponent<LayoutElement>().preferredWidth = 100;
+        }
+
+        private void UpdateLayout(string currentCombination)
+        {
+            var childCount = _layoutGroup.transform.childCount;
+            for (var i = childCount - 1; i >= 0; i--)
+            {
+                GameObject.Destroy(_layoutGroup.transform.GetChild(i).gameObject);
+            }
+            var individualCombos = currentCombination.Split('/');
+            for (var i = 0; i < individualCombos.Length; i++)
+            {
+                var keyStrings = individualCombos[i].Split('+');
+                for (var j = 0; j < keyStrings.Length; j++)
+                {
+                    AddKeySign(keyStrings[j]);
+                    if (j < keyStrings.Length - 1)
+                    {
+                        Button.AddText("+");
+                    }
+                }
+                if (i < individualCombos.Length - 1)
+                {
+                    Button.AddText("/");
+                }
+            }
+            Button.UpdateState();
+        }
+
+        private Texture2D GetGamepadButtonTexture(string key)
+        {
+            return ButtonPromptLibrary.SharedInstance.GetButtonTexture((JoystickButton)Enum.Parse(typeof(JoystickButton), key));
+        }
+
+        private Texture2D GetXboxButtonTexture(string xboxKey)
+        {
+            switch (xboxKey[0])
+            {
+                case 'A':
+                    return ButtonPromptLibrary.SharedInstance.GetButtonTexture(JoystickButton.FaceDown);
+                case 'B':
+                    return ButtonPromptLibrary.SharedInstance.GetButtonTexture(JoystickButton.FaceRight);
+                case 'X':
+                    return ButtonPromptLibrary.SharedInstance.GetButtonTexture(JoystickButton.FaceLeft);
+                case 'Y':
+                    return ButtonPromptLibrary.SharedInstance.GetButtonTexture(JoystickButton.FaceUp);
+                default:
+                    return GetGamepadButtonTexture(xboxKey);
+            }
+        }
+
+        private void AddKeySign(string key)
+        {
+            Button.AddPicture(
+                key.Contains(XboxPrefix) ?
+                GetXboxButtonTexture(key.Substring(XboxPrefix.Length)) :
+                ButtonPromptLibrary.SharedInstance.GetButtonTexture((KeyCode)Enum.Parse(typeof(KeyCode), key))
```

Comment:
this should be easier to read if you make a variable for the texture

Reply:
with ModInputLibrary it's somewhat better, cos that's just method call now. But sure why not

Reply:
done